falcon/request.py

```diff
@@ -1096,6 +1112,20 @@ def _parse_form_urlencoded(self):
 
             self._params.update(extra_params)
 
+    def _parse_application_json(self):
+        raw_body = self.stream.read()
+
+        body = None
+        try:
+            body = json.loads(raw_body.decode('utf-8'))
+        except UnicodeDecodeError:
+            self.log_error('Non-UTF8 characters found in the request body')
+        except ValueError as e:
+            msg = 'Could not parse the body as Json: {0}. Ignoring.'.format(e)
+            self.log_error(msg)
+
+        self._json = body
```

Comment:
This method should not set attribute but only return value and I think that we should avoid such side effects if there is no clue about that in method name or at least docstring (missing btw).

Where to store value it is a responsibility of the method caller caller.